ws/redisPubSub.js

```diff
@@ -0,0 +1,43 @@
+const config = require('../config')
+var ws = require('./sockets');
+var redis = require('redis')
+    , subscriber = redis.createClient(config.redisStore.url)
+    , r = redis.createClient(config.redisStore.url)
+
+clients = ws.clients;
+
+redisPSManager = function(socket) {
+    deviceIds = []
+    socket.on("disconnect", function() {
+        console.log('disconnected from socket!');
+        console.log('unsubscribing from deviceIds: ' + deviceIds);
+        for( var i = 0; i < deviceIds.length; i+=1) {
+            sub.unsubscribe(deviceIds[i]);
+        }
+    });
+
+    var sub = redis.createClient(config.redisStore.url);
+    sub.on("subscribe", function(channel, count) {
+        console.log("Subscribed to " + channel + ". Now subscribed to " + count + " channel(s).");
+    });
+
+    sub.on('message', function(channel, jsonString) {
+        data = JSON.parse(jsonString);
+        console.log(data);
+        socket.emit('chartData', {
```

Comment:
I don't think we should unpack this. just send the data object from above and edit chart test accordingly for now.